---
bool is_max_reached
